package/MDAnalysis/analysis/base.py

```diff
@@ -125,14 +125,192 @@
 import inspect
 import logging
 import itertools
+import warnings
 
 import numpy as np
 from MDAnalysis import coordinates
 from MDAnalysis.core.groups import AtomGroup
 from MDAnalysis.lib.log import ProgressBar
+from MDAnalysis.lib.util import is_installed
+from typing import Callable, Iterable
+from importlib import import_module
+from functools import partial
 
 logger = logging.getLogger(__name__)
 
+def is_valid_distributed_client(client: object):
+    try:
+        import dask.distributed as distributed
+    except ImportError:
+        return False
+    return isinstance(client, distributed.Client)
+
+class ParallelExecutor:
+    r"""Executor object that abstracts away running parallel computations
+    with various backends. Mainly is used for AnalysisBase, but can be used for other purposes as well."""
+
+    @classmethod
+    @property
+    def available_backends(cls):
+        """Declares backends that are implemented for parallel execution.
+        Importantly, backend names are the same as packages you must import
+        in order to run them -- e.g. to run 'dask.distributed', you need to run 'import dask.distributed'
+
+        Returns
+        -------
+        available_backends : list[str]
```

Comment:
```suggestion
        list[str]
```

I think return documentation has no name? (IIRC)

Reply:
yes, you're right. Fixed.